### PR TITLE
Make clear when to use authorization code vs client credentials flow

### PR DESCRIPTION
**Pull request by turnipdabeets**

### CODE REVIEW

**Review by Will5 — APPROVED** [review]
LGTM

File: docs/workflows/applications.md [hunk 1]
```diff
@@ -80,7 +80,7 @@ An application can also be created from the [Beyond Identity API](https://develo
 
 Before creating an application by API, you may want to create a resource server. A resource server is a namespace for application scopes. Application scopes is a set of all scopes supported by the application. When an application is created without a resource server, this application may provide authentication (identity) but not authorization (access). If your application doesn't provide multiple levels of access -- like admin access vs user access, then there might not be a need for a resource server.
 
-When creating an application with the [admin console](#create-application-from-admin-console), if the application lives in the admin realm, then the application is created with the management api resource server. Otherwise, it is currently created without a resource server.
+When creating an application with the [admin console](#create-application-from-admin-console), if the application lives in the admin realm, then the application is created with the management API resource server. Otherwise, it is currently created without a resource server.
```
Comment (tomsik68) [inline docs/workflows/applications.md:83]: > if the application lives in the admin realm, then the application is created with the management API resource server

I might be missing some context here, but I don't agree with this statement. In console, the user needs to explicitly select the Management API resource server. From curl, the user needs to explicitly attach the resource server id to the json request body. We never attach an application to a resource server.

Comment (turnipdabeets) [reply docs/workflows/applications.md:83]: Oh wow, at the time of writing this field didn't exist. I guess it was recently added? I might need to update the applications documentation too.

Screenshot 2023-03-02 at 12 28 18 PM

Comment (turnipdabeets) [reply docs/workflows/applications.md:83]: But what happens if you select "None"?

Comment (tomsik68) [reply docs/workflows/applications.md:83]: Apart from the resource server field, LGTM! We might want to address that in separate PR so I'm approving.

**Pull request merged**